Shipping Labels: Better formatted surcharge for additional shipping services

itsmeichigo opened the pull request:
<!-- Remember about a good descriptive title. -->

<!-- Id number of the Linear issue this PR addresses. -->
<!-- Part of: # -> use this if your PR is one of many related to one issue -->

Part of WOOMOB-683

## Description
<!-- Take the time to write a good summary. Why is it needed? What does it do? When fixing bugs try to avoid just writing “See original issue” – clarify what the problem was and how you’ve fixed it. -->

I found an edge case where the surcharges for the add-ons are negative:

<img src="https://github.com/user-attachments/assets/b77e9938-853d-4686-9739-1d358eca217e" width=320 />


This PR updates the formatting of surcharges for addtional shipping services to account for negative values. This is most likely an edge case but the fix gives us more confidence in the code.

Also, I removed the added alignment for the HStacks for a better layout. The addtional handling service is also moved above Saturday delivery to match with the plugin.

Additionally, a small fix was added to mitigate the issue purchasing a label after accepting UPS TOS. This issue came from resetting `selectedRate` causing the refreshing of the rate after accepting TOS to fail. The solution is to keep hold of the rate before refreshing packages and rates.

## Testing steps
<!-- Step-by-step testing instructions. For new user flows, consider instead stating the goal of the workflow and see if your PR reviewer can accomplish the workflow without specific steps! -->

1. Log in to a test store with Woo Shipping plugin set up.
2. Navigate to the Orders tab and select a paid order with physical products.
3. Select Create shipping labels > Add a package > select a UPS package.
4. Select any shipping rates and confirm that the layout looks good and Addtional Handling is above Saturday Delivery.
5. The negative surcharges are likely a bug on the backend side and not easy to reproduce. Update the response of `POST /wcshipping/v1/label/rate` to make the rates of additional services lower than the standard rates.
6. Reload the rates and confirm that the negative values are displayed correctly.
7. Change the origin address to a new one that hasn't accepted UPS TOS.
8. Proceed to purchase a label. Confirm that after accepting UPS TOS, the purchase succeeds.

## Testing information
<!-- This is your opportunity to break out individual scenarios that need testing (when necessary) and/or include a checklist for the reviewer to go through. Consider documenting the following from your own completed testing: devices used, alternate workflows, edge cases, affected areas, critical flows, areas not tested, and any remaining unknowns. Provide feedback on this new section of the PR template through Sept 30, 2024 to Apps Quality; additional context here: https://woomobilep2.wordpress.com/2024/05/06/woocommerce-mobile-quality-report-march-april/#comment-12036 -->

- Tested and confirmed with simulator iPhone 16 iOS 18.4.
- Added unit test for the negative value case.

## Screenshots
<!-- Include before and after images or gifs when appropriate. -->

<img src="https://github.com/user-attachments/assets/903c890b-f8d3-44d2-af54-1b044cb07749" width=320 />


---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

itsmeichigo committed:
Update formatted surcharge for add-ons to handle negative values

itsmeichigo committed:
Move additional handling to above saturday delivery

dangermattic commented:
<!--
  0 Errors
  1 Warning: This PR is assigned to the mil...
  0 Messages
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/woocommerce/woocommerce-ios/milestone/247">22.7</a>. This milestone is due in less than 2 days.<br />
Please make sure to get it merged by then or assign it to a milestone with a later deadline.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

itsmeichigo committed:
Fix issue refreshing packages and shipping rates after accepting TOS

A previous PR resets the selectedRate after a new package is selected. Here I persist the selectedRate before refreshing to use for refreshing rates later.

itsmeichigo commented on WooCommerce/Classes/ViewRelated/Orders/Order Details/Shipping Labels/WooShipping Create Shipping Labels/ShipmentDetails/WooShippingShipmentDetailsViewModel.swift:
```diff
@@ -174,7 +174,7 @@ final class WooShippingShipmentDetailsViewModel: ObservableObject {
     @MainActor
     func refreshPackagesAndShippingRates() async throws {
-        guard let selectedPackage, let updatedPackage = try await refreshSelectedPackage(from: selectedPackage) else {
+        guard let selectedRate, let selectedPackage, let updatedPackage = try await refreshSelectedPackage(from: selectedPackage) else {
```
A previous PR resets the selectedRate after a new package is selected. Here I persist the selectedRate before refreshing to use for refreshing rates later.

itsmeichigo committed:
Update unit tests

RafaelKayumov reviewed (approved):
LGTM. Tested. Works as described.
